providers/core/core.unprivileged.js

```diff
@@ -185,6 +188,45 @@ Core_unprivileged.prototype.bindChannel = function(identifier, continuation, sou
 };
 
 /**
+ * Require a dynamic dependency for your freedom module.
+ * If new permissions are needed beyone what are already available to the
+ * freedom context, the user will need to approve of the requested permissions.
+ * @method require
+ * @param {String} manifest The URL of the manifest to require.
+ * @param {String} api The API of the dependency to expose if not default.
+ * @param {Function} callback The function to call with the dependency.
+ */
+Core_unprivileged.prototype.require = function (manifest, api, callback) {
+  if (this.manager.delegate && this.manager.toDelegate.core &&
+      Core_unprivileged.moduleInternal) {
+    // Register a callback with moduleInternal.
+    // DependencyName is the name of the channel moduelInternal will allocate
+    // callback will be called once a link to that channel is seen.
+    var dependencyName =
+        Core_unprivileged.moduleInternal.registerId(api, callback);
+
+    // Request the dependency be added.
+    this.manager.emit(this.manager.delegate, {
+      type: 'Delegation',
+      request: 'handle',
+      flow: 'core',
+      message: {
+        type: 'require',
+        manifest: manifest,
+        id: dependencyName
+      }
+    });
+  } else {
+    this.debug.error('The require function in external context makes no sense' +
+        ' Instead create a new freedom() context.');
+    callback(undefined, {
+      errcode: 'InvalidContext',
```

Comment:
Does it make sense to add this error to core.json, since it is related to the API specified there?